gobblin-service/src/test/java/org/apache/gobblin/service/modules/core/GobblinServiceHATest.java

```diff
@@ -546,6 +551,30 @@ public void testKillNode() throws Exception {
 
     Assert.assertTrue(assertSuccess, "New master should take over all old master jobs.");
 
+    // Check eventbus was registered with new leader
+    attempt = 0;
+    assertSuccess = false;
+    while (attempt < 10) {
+      try {
+        Mockito.verify(secondary.getEventBus(), Mockito.atLeastOnce()).register(secondary.getDagManager());
+        assertSuccess = true;
+      } catch (MockitoAssertionError e) {
+        logger.info("Verification failed", e);
+      }
+      Thread.sleep(5);
```

Comment:
The test makes 10 attempts with 5ms intervals. Is 50ms enough for the switch to happen on a busy test machine with high CPU usage? Looks like this can be a source of flakiness for the test.

Reply:
Didn't notice it was ms, but changed to assertwithbackoff with 20 second timeout